overlord/hookstate/ctlcmd/refresh.go

```diff
@@ -0,0 +1,162 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/snap"
+)
+
+type refreshCommand struct {
+	baseCommand
+
+	Pending bool `long:"pending" description:"Show pending refreshes of the calling snap"`
+	// these two options are mutually exclusive
+	Proceed bool `long:"proceed" description:"Proceed with potentially disruptive refreshes"`
+	Hold    bool `long:"hold" description:"Do not proceed with potentially disruptive refreshes"`
+}
+
+var shortRefreshHelp = i18n.G("The refresh command prints pending refreshes and can hold disruptive ones.")
+var longRefreshHelp = i18n.G(`
+The refresh command prints pending refreshes of the calling snap and can hold
+disruptive refreshes of other snaps, such as refreshes of kernel or
+base snaps that can trigger restart. This command can be used from
+gate-auto-refresh hook. The hook is only run during auto-refresh.
+
+Snap can query pending refreshes with:
+    $ snapctl refresh --pending
+    pending: ready
+    channel: stable
+    version: 2
+    revision: 2
+    base: false
+    restart: false
+
+The 'pending' flag can be "ready", "none" or "inhibited". It is set to "none" if
+there are no pending refreshes for the snap. It is set to "ready" if there are,
+and to inhibited if there are but are inhibited because some of the snap
+applications are running and "refresh app awareness" feature is enabled.
+The "base" and "restart" flags indicate whether a base snap is going to be
+updated and/or if restart will occur, both of which are disruptive.
+
+To tell snapd to proceed with pending refreshes:
+    $ snapctl refresh --pending --proceed
+
+Note, a snap using --proceed cannot assume that the updates will occur as they
+might be held by other snaps.
+
+To hold refresh for up to 90 days:
```

Comment:
this will need to be clarified, it's 90 days for holding the snap itself but not for holding other snaps